easybuild/easyconfigs/s/Salmon/Salmon-1.2.0-gompi-2019b.eb

```diff
@@ -0,0 +1,49 @@
+# #
+# This is a contribution from HPCNow! (http://hpcnow.com)
+# Copyright::   HPCNow!
+# Authors::     Jordi Blasco <jordi.blasco@hpcnow.com>
+# License::     GPL-v3.0
+# Based on Robert Qiao <rob.qiao@flinders.edu.au> contribution
+# #
+
+easyblock = 'CMakeMake'
+
+name = 'Salmon'
+version = '1.2.0'
+
+homepage = 'https://salmon.readthedocs.io'
+description = """Salmon is a wicked-fast program to produce a highly-accurate,
+transcript-level quantification estimates from RNA-seq data. Salmon achieves
+its accuracy and speed via a number of different innovations, including the use
+of selective-alignment, and massively-parallel stochastic collapsed variational
+inference."""
+
+toolchain = {'name': 'gompi', 'version': '2019b'}
+toolchainopts = {'pic': True, 'openmp': True}
+
+github_account = 'COMBINE-lab'
+source_urls = ['https://github.com/%(github_account)s/%(namelower)s/archive/']
+sources = ['v%(version)s.tar.gz']
+checksums = ['c8a6647484ff630e4958aa4540cba28c0913b26922313876d747aba65eefb96f']
+
+builddependencies = [
+    ('CMake', '3.15.3'),
+    ('pkg-config', '0.29.2'),
+    ('jemalloc', '5.2.1'),
+    ('Boost', '1.71.0'),
+    ('zlib', '1.2.11'),
+    ('bzip2', '1.0.8'),
+]
+
+dependencies = [
+    ('tbb', '2019_U9'),
+]
+
```

Comment:
Could you clarify why the `configopts` and `pretestopts` from Salmon v1.1.0 have been removed? I think that by removing them this easyconfig would fail to use `jemalloc` from EB.

Reply:
The -DBOOST_ROOT, -DTBB_INSTALL_DIR, -DJEMALLOC_ROOT, etc. are no longer required. The https://github.com/COMBINE-lab/salmon/blob/master/CMakeLists.txt has enough logic to identify the right libraries exposed by EasyBuild.

Reply:
Please, have a look at the ldd output and the compilation options:

````
scicomp@6c650a6a60f7:/fredhutch$ ml

Currently Loaded Modules:
  1) EasyBuild/4.1.1               6) numactl/2.0.12-GCCcore-8.3.0     11) OpenMPI/3.1.4-GCC-8.3.0
  2) GCCcore/8.3.0                 7) XZ/5.2.4-GCCcore-8.3.0           12) gompi/2019b
  3) zlib/1.2.11-GCCcore-8.3.0     8) libxml2/2.9.9-GCCcore-8.3.0      13) tbb/2019_U9-GCCcore-8.3.0
  4) binutils/2.32-GCCcore-8.3.0   9) libpciaccess/0.14-GCCcore-8.3.0  14) Salmon/1.2.0-gompi-2019b
  5) GCC/8.3.0                    10) hwloc/1.11.12-GCCcore-8.3.0



scicomp@6c650a6a60f7:/fredhutch$ ldd /app/software/Salmon/1.2.0-gompi-2019b/bin/salmon
	linux-vdso.so.1 (0x00007ffe570f4000)
	libtbbmalloc_proxy.so.2 => /app/software/tbb/2019_U9-GCCcore-8.3.0/build/linux_intel64_gcc_cc8.3.0_libc2.27_kernel4.15.0_release/libtbbmalloc_proxy.so.2 (0x00007fc7f6b73000)
	libtbbmalloc.so.2 => /app/software/tbb/2019_U9-GCCcore-8.3.0/build/linux_intel64_gcc_cc8.3.0_libc2.27_kernel4.15.0_release/libtbbmalloc.so.2 (0x00007fc7f6b35000)
	libtbb.so.2 => /app/software/tbb/2019_U9-GCCcore-8.3.0/build/linux_intel64_gcc_cc8.3.0_libc2.27_kernel4.15.0_release/libtbb.so.2 (0x00007fc7f6af3000)
	libgomp.so.1 => /app/software/GCCcore/8.3.0/lib64/libgomp.so.1 (0x00007fc7f6ac2000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fc7f674b000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fc7f6547000)
	libstdc++.so.6 => /app/software/GCCcore/8.3.0/lib64/libstdc++.so.6 (0x00007fc7f63ad000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fc7f600f000)
	libgcc_s.so.1 => /app/software/GCCcore/8.3.0/lib64/libgcc_s.so.1 (0x00007fc7f6aa5000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fc7f5df0000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fc7f59ff000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fc7f6953000)
```

````
/app/software/GCCcore/8.3.0/bin/g++  -O2 -ftree-vectorize -march=native -fno-math-errno -fopenmp -fPIC -O3 -DNDEBUG -flto -fno-fat-lto-objects  -L/app/software/bzip2/1.0.8-GCCcore-8.3.0/lib -L/app/software/zlib/1.2.11-GCCcore-8.3.0/lib -L/app/software/Boost/1.71.0-gompi-2019b/lib -L/app/software/jemalloc/5.2.1-GCCcore-8.3.0/lib -L/app/software/GCCcore/8.3.0/lib64 -L/app/software/GCCcore/8.3.0/lib CMakeFiles/salmon.dir/EMUtils.cpp.o CMakeFiles/salmon.dir/CollapsedEMOptimizer.cpp.o CMakeFiles/salmon.dir/CollapsedCellOptimizer.cpp.o CMakeFiles/salmon.dir/CollapsedGibbsSampler.cpp.o CMakeFiles/salmon.dir/Salmon.cpp.o CMakeFiles/salmon.dir/BuildSalmonIndex.cpp.o CMakeFiles/salmon.dir/Graph.cpp.o CMakeFiles/salmon.dir/DedupUMI.cpp.o CMakeFiles/salmon.dir/Alevin.cpp.o CMakeFiles/salmon.dir/AlevinHash.cpp.o CMakeFiles/salmon.dir/SalmonAlevin.cpp.o CMakeFiles/salmon.dir/WhiteList.cpp.o CMakeFiles/salmon.dir/SalmonQuantify.cpp.o CMakeFiles/salmon.dir/FragmentLengthDistribution.cpp.o CMakeFiles/salmon.dir/FragmentStartPositionDistribution.cpp.o CMakeFiles/salmon.dir/GZipWriter.cpp.o CMakeFiles/salmon.dir/SalmonQuantMerge.cpp.o CMakeFiles/salmon.dir/ProgramOptionsGenerator.cpp.o CMakeFiles/salmon.dir/FASTAParser.cpp.o CMakeFiles/salmon.dir/AlignmentModel.cpp.o CMakeFiles/salmon.dir/SalmonQuantifyAlignments.cpp.o CMakeFiles/salmon.dir/BAMUtils.cpp.o  -o salmon  -L/app/build/Salmon/1.2.0/gompi-2019b/salmon-1.2.0/lib  -L/app/build/Salmon/1.2.0/gompi-2019b/salmon-1.2.0/external/install/lib -Wl,-rpath,"\$ORIGIN/../lib:\$ORIGIN/../../lib:\$ORIGIN/:\$ORIGIN/../../external/install/lib" ../external/pufferfish/src/libpuffer.a libsalmon_core.a ../external/pufferfish/external/twopaco/graphconstructor/libtwopaco.a ../external/pufferfish/external/twopaco/graphdump/libgraphdump.a ../external/pufferfish/external/ntcard/libntcard.a -lgff /app/software/Boost/1.71.0-gompi-2019b/lib/libboost_iostreams.a /app/software/Boost/1.71.0-gompi-2019b/lib/libboost_filesystem.a /app/software/Boost/1.71.0-gompi-2019b/lib/libboost_system.a /app/software/Boost/1.71.0-gompi-2019b/lib/libboost_timer.a /app/software/Boost/1.71.0-gompi-2019b/lib/libboost_chrono.a /app/software/Boost/1.71.0-gompi-2019b/lib/libboost_program_options.a /app/software/Boost/1.71.0-gompi-2019b/lib/libboost_regex.a ../external/install/lib/libstaden-read.a /app/software/cURL/7.66.0-GCCcore-8.3.0/lib/libcurl.a /app/software/zlib/1.2.11-GCCcore-8.3.0/lib/libz.a -lm /app/software/XZ/5.2.4-GCCcore-8.3.0/lib/liblzma.a /app/software/bzip2/1.0.8-GCCcore-8.3.0/lib/libbz2.a -ltbbmalloc_proxy -ltbbmalloc -ltbb -lgomp /app/software/jemalloc/5.2.1-GCCcore-8.3.0/lib/libjemalloc.a -lrt ../external/pufferfish/src/libksw2pp.a libalevin_core.a -ldl
```

Reply:
You are right it now detects `jemalloc` automatically:
```
-- Found Jemalloc: /apps/brussel/CO7/ivybridge-ib/software/jemalloc/5.2.1-GCCcore-8.3.0/lib/libjemalloc.a (found version "5.2.1-0-g0000"     )                                                                                                                                       
Found Jemalloc library --- using this memory allocator
```